servicetalk-concurrent-api/src/testFixtures/java/io/servicetalk/concurrent/api/TestCompletable.java

```diff
@@ -148,7 +148,7 @@ public void onError(final Throwable t) {
 
     private Subscriber checkSubscriberAndExceptions() {
         if (!exceptions.isEmpty()) {
-            final RuntimeException exception = new RuntimeException("Unexpected exception(s) encountered",
+            final IllegalStateException exception = new IllegalStateException("Unexpected exception(s) encountered",
```

Comment:
If it's a test utility, should we use `AssertionError` in this case for `TestCompletable`, `TestSingle`, and `TestPublisher`?